src/test/httpserver_tests.cpp

```diff
@@ -34,5 +34,9 @@ BOOST_AUTO_TEST_CASE(test_query_parameters)
     // Invalid query string syntax is the same as not having parameters
     uri = "/rest/endpoint/someresource.json&p1=v1&p2=v2";
     BOOST_CHECK(!GetQueryParameterFromUri(uri.c_str(), "p1").has_value());
+
+    // Invalid query string with invalid will raise a runtime error no matter how many parameters passed or their syntax
```

Comment:
I'd rephrase to the below - any invalid character in the URI will trigger this, not just in the query string.

```
    // URI with invalid characters (%) raises a runtime error regardless of which query parameter is queried
```